src/frontend/js/settings/index.ts

```diff
@@ -0,0 +1,33 @@
+import { isArray, mergeWith } from 'lodash-es';
+import prodSettings from './settings.prod';
+
+let localSettings;
+try {
+  localSettings = require('./settings.local.ts');
+} catch {
+  localSettings = {};
+}
+
+const settings = !['production', 'test'].includes(process.env.NODE_ENV || 'prod')
+  ? mergeWith({}, prodSettings, localSettings.default, (_objValue, sourceObj) => {
+      if (isArray(sourceObj)) {
+        return sourceObj;
+      }
+    })
+  : prodSettings;
```

Comment:
Why do you need to use a `customizer` to check if the sourceObj is an array ?
For our case, I think the `merge` method does the job nope ?

Furthermore, why not choose `prodSettings` has `source` ?

Reply:
`customizer` allow use to replace arrays instead of merge them.
With:
prod: `MY_ARRAY_SETTING: ['a', 'b']`
and local: `MY_ARRAY_SETTING: ['c']`
We want the final result to be `MY_ARRAY_SETTING: ['c']` and not `MY_ARRAY_SETTING: ['a', 'b', 'c']`

The first obj of lodash merge function will be modify like if we used `assign`. I feel more comfortable not to change `prodSettings` and `localSettings` objects